### PR TITLE
[Setup] Firebase - Crashlytics, analytics, performance and Cloud Messaging

### PR DESCRIPTION
**Pull request by kibettheophilus**
# Scope

_Please make sure to read https://github.com/droidconKE/droidconKE2022Android/docs/CONTRIBUTING.md
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
Closes - [Setup Firebase](https://trello.com/c/7eOe7TPf/42-setup-firebase-crashlytics-analytics-performance-and-cloud-messaging)
## Other testing QA Notes
N/A

### CODE REVIEW

File: .github/workflows/branch.yml [hunk 1]
```diff
@@ -73,6 +73,9 @@ jobs:
           subdirectory: fastlane
           bundle-install-path: CACHE_BUNDLER
 
+      - name: Create Google Services Json
+        run: echo '${{ secrets.GOOGLE_SERVICES_JSON }}' > /home/runner/work/droidconKE2022Android/droidconKE2022Android/app/google-services.json
+
```
Comment (chepsi) [inline .github/workflows/branch.yml:78]: Get rid of this we will create a new workflow for deployments.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -47,6 +47,9 @@ jobs:
       - name: Install Dependencies
         run: gem install bundler && bundle install
 
+      - name: Create Google Services Json
+        run: echo '${{ secrets.GOOGLE_SERVICES_JSON }}' > /home/runner/work/droidconKE2022Android/droidconKE2022Android/app/google-services.json
+
```
Comment (chepsi) [inline .github/workflows/main.yml:52]: Same as above.

File: app/src/main/java/com/android254/droidconKE2022/services/MessagingService.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+class MessagingService : FirebaseMessagingService() {
+    override fun onNewToken(token: String) {
+        super.onNewToken(token)
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        // Check if message contains a data payload.
```
Comment (chepsi) [inline app/src/main/java/com/android254/droidconKE2022/services/MessagingService.kt:37]: Remove unnecessary comment here.

**Review by chepsi — APPROVED** [review]
LGTM

File: .github/workflows/branch.yml (hunk 1 above)
Comment (kibettheophilus) [reply .github/workflows/branch.yml:78]: If I remove this then the build will fail, the other way is to commit the **google-services.json** file which I didn't want to.

File: .github/workflows/main.yml (hunk 2 above)
Comment (kibettheophilus) [reply .github/workflows/main.yml:52]: Answered above

**Review by jumaallan — APPROVED** [review]

**Pull request merged**